api/v1beta1/bgppeer_types.go

```diff
@@ -81,6 +82,10 @@ type BGPPeerSpec struct {
 	// +optional
 	Password string `json:"password,omitempty" yaml:"password,omitempty"`
 
+	// secretRef is name of the authentication secret for BGP Peer
```

Comment:
secretRef is the name or a reference? i'm still not sure how it's going to look like

Reply:
changed it to use `corev1.SecretReference `and added few unit tests. hope that explains it clearly.